packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -0,0 +1,314 @@
+/** @import { Derived, Effect, Source } from '#client' */
+import { CLEAN, DIRTY } from '#client/constants';
+import { deferred } from '../../shared/utils.js';
+import {
+	flush_queued_effects,
+	flush_queued_root_effects,
+	process_effects,
+	schedule_effect,
+	set_queued_root_effects,
+	set_signal_status,
+	update_effect
+} from '../runtime.js';
+
+/** @type {Set<Batch>} */
+const batches = new Set();
+
+/** @type {Batch | null} */
+export let current_batch = null;
+
+/** @type {Map<Derived, any> | null} */
+export let batch_deriveds = null;
+
+/** TODO handy for debugging, but we should probably eventually delete it */
+let uid = 1;
+
+export class Batch {
```

Comment:
This class would _really_ benefit from good JSDoc, especially given its reach across the codebase, _especially_ on complex methods like `process`. I can see what the code _does_ but trying to intuit _why_ is quite challenging

Reply:
I've added a bunch of JSDoc comments around the less self-explanatory stuff. No doubt there's more that could be added but I'll mark this resolved for now